lib/beaker/subcommand.rb

```diff
@@ -6,15 +6,41 @@ module Beaker
   class Subcommand < Thor
     SubcommandUtil = Beaker::Subcommands::SubcommandUtil
 
+    def initialize(*args)
+      super
+      @@config = SubcommandUtil.load_config()
+    end
+
     desc "init", "Initialises the beaker test environment configuration"
     option :hypervisor, :type => :string, :required => true
     long_desc <<-LONGDESC
-      Initialises a beaker environment configuration
+    Initialises a beaker environment configuration
     LONGDESC
     def init()
       SubcommandUtil.verify_init_args(options)
       SubcommandUtil.require_tasks()
       SubcommandUtil.init_hypervisor(options)
+      SubcommandUtil.store_config(options)
+      SubcommandUtil.delete_config([:provisioned, :hosts])
```

Comment:
We can talk more around this, but my thinking that a call to init should *require* provisioning to run. To achieve this, I delete any config written by provision at the end of the init run.